store/iavl/store.go

```diff
@@ -273,7 +273,13 @@ func (st *Store) Export(version int64) (*iavl.Exporter, error) {
 	if !ok || tree == nil {
 		return nil, fmt.Errorf("iavl export failed: unable to fetch tree for version %v", version)
 	}
-	return tree.Export(), nil
+	exporter := tree.Export()
+
+	// cosmos/iavl immutable_tree.go and export.go FAIL if tree.ndb is nil
+	if exporter == nil {
+		return nil, fmt.Errorf("iavl export failed: unable to create exporter")
```

Comment:
This would mean the snapshot is unusable, it might make sense to not complete the snapshot at all in this case

Reply:
Yes. With the additional real world usage we have seen, it is better to remove this change and fail the snapshot.